File: content/snippets/go/interview_lcci_05_01_insert.go
package main

import "fmt"

func insert(N int, M int, i int, j int) int {
	mask := ^(((1 << (j - i + 1)) - 1) << i)
	return (N & mask) | (M << i)
}

func main() {
	fmt.Println(insert(1024, 19, 2, 6))
	fmt.Println(insert(0, 19, 0, 4))
}

File: content/snippets/scala3/interview_lcci_05_01_insert.scala
object Lcci0501Insert:
  def insert(N: Int, M: Int, i: Int, j: Int): Int =
    val mask = ~(((1 << (j - i + 1)) - 1) << i)
    (N & mask) | (M << i)

  @main def main(): Unit =
    println(insert(1024, 19, 2, 6))
    println(insert(0, 19, 0, 4))
